server/src/main/java/org/elasticsearch/search/aggregations/support/BuiltinValuesSourceType.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.support;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.fielddata.IndexFieldData;
+import org.elasticsearch.index.fielddata.IndexGeoPointFieldData;
+import org.elasticsearch.index.fielddata.IndexNumericFieldData;
+import org.elasticsearch.index.fielddata.IndexOrdinalsFieldData;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.RangeFieldMapper;
+import org.elasticsearch.script.AggregationScript;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.aggregations.AggregationExecutionException;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.function.LongSupplier;
+
+public enum BuiltinValuesSourceType implements Writeable, ValuesSourceType {
```

Comment:
"Built in" meaning like the implicit types available in core?

Reply:
Correct.  I'm open to other names, but that's the best I could come up with.

Reply:
sorry for the late fly-by. I have a history of poor naming, so please take any suggestion I have with a grain of salt....

I believe we've historically distinguished between the things in server as opposed to modules/plugins by calling it "core". So maybe `CoreValuesSourceType`? Builtin suggests it would not come pre-loaded in a distribution of ES, but if we have types as modules, then both would be built-in, in a way.

Reply:
I like "core" as well.  ++